### DISCUSSION TITLE
Encryption of the API token

### DISCUSSION THREAD

**Discussion by Moepeldimoep** [pterodactyl/panel#2969] [Support] [answered]
Hello there!
How does the API key get generated from the token saved in the database?
Example:
Identifier: 123
Token: otz2134pda35uDhgGJHi2134uasdhgfpiuziu (Something encoded in Base64)
Key: 123abc

So how do i get from "otz2134pda35uDhgGJHi2134uasdhgfpiuziu" to "abc"?

**Comment by Kaesewuerfel** [comment]
same problem

**Comment by schrej** [comment, marked as answer]
The token that you see in the database is encrypted using the `APP_KEY` defined in the `.env` file. It can be decrypted using the `decrypt()` method of the `Encrypter` that is part of laravel, see [here](https://github.com/pterodactyl/panel/blob/develop/app/Http/Middleware/Api/AuthenticateKey.php#L109) for an example.

**Reply by Kaesewuerfel** [reply]
Sänk you so musch
